### PR TITLE
Aerogear 3133 fail gracefully

### PR DESCRIPTION
**Pull request by secondsun**
## Motivation

This makes the application not crash when mobile services are not configured and has a toast pop up when you try to navigate to a service 

https://issues.jboss.org/browse/AEROGEAR-3133## Description

## Progress

- [x] Auth Service Won't Crash
- [x] Metrics Service Won't Crash
- [x] Notes Service Won't Crash
- [x] Friendly Toast

## Additional Notes

This is not the finished work with the friendly dialog box, that work is tracked under the jira : https://issues.jboss.org/browse/AEROGEAR-3060

### CODE REVIEW

**Comment by rachael-oregan** [comment]
@secondsun Looks good.  Verified this works by removing the services one by one.  Also works with no services defined:

screen shot 2018-06-13 at 13 29 08

**Comment by secondsun** [comment]
Awesome, I need to send an update to this, I was using getconfigbyid when I should use getconfigbytype.

File: app/src/main/java/com/aerogear/androidshowcase/di/SecureApplicationModule.java [hunk 1]
```diff
@@ -107,9 +108,13 @@ SecurityService provideSecurityService() {
     // end::securityServiceInit[]
 
     // tag::authServiceInit[]
-    @Provides @Singleton
+    @Provides @Singleton @Nullable
     AuthService provideAuthService(Context context) {
-        AuthService authService = MobileCore.getInstance().getService(AuthService.class);
+        MobileCore core = MobileCore.getInstance();
+        if (core.getServiceConfigurationById("keycloak") == null ) {
+            return null;//I don't like null here, but we don't have the ability to have an
```
Comment (danielpassos) [inline app/src/main/java/com/aerogear/androidshowcase/di/SecureApplicationModule.java]: an ... ?

**Comment by secondsun** [comment]
Ok, I've replaced id by type, @danielpassos can you :eye:  it again?

File: app/src/main/java/com/aerogear/androidshowcase/di/SecureApplicationModule.java [hunk 2]
```diff
@@ -107,9 +108,13 @@ SecurityService provideSecurityService() {
     // end::securityServiceInit[]
 
     // tag::authServiceInit[]
-    @Provides @Singleton
+    @Provides @Singleton @Nullable
     AuthService provideAuthService(Context context) {
-        AuthService authService = MobileCore.getInstance().getService(AuthService.class);
+        MobileCore core = MobileCore.getInstance();
+        if (core.getServiceConfigurationByType("keycloak") == null ) {
+            return null;//I don't like null here, but we don't have the ability to have an
```
Comment (danielpassos) [inline app/src/main/java/com/aerogear/androidshowcase/di/SecureApplicationModule.java]: an ... ?

Comment (secondsun) [reply app/src/main/java/com/aerogear/androidshowcase/di/SecureApplicationModule.java]: Fixed

**Pull request merged**